woof-code/support/cros_image.sh

```diff
@@ -0,0 +1,152 @@
+              --arch $WOOF_TARGETARCH \
+              --keyblock /usr/share/vboot/devkeys/kernel.keyblock \
+              --signprivate /usr/share/vboot/devkeys/kernel_data_key.vbprivk \
+              --config cmdline \
+              --bootloader bootloader.bin
+
+mkdir -p /mnt/sdimagep2 /mnt/ssdimagep2
+
+create_image() {
+	# it's a sparse file - that's how we fit a 16GB image inside a 2GB one
+	dd if=/dev/zero of=$1 bs=$2 count=$3 conv=sparse
+	parted --script $1 mklabel gpt
+	cgpt create $1
+	cgpt add -i 1 -t kernel -b 8192 -s 65536 -l Kernel -S 1 -T 5 -P 10 $1
+	start=$((8192 + 65536))
+	end=`cgpt show $1 | grep 'Sec GPT table' | awk '{print $1}'`
+	size=$(($end - $start))
+	cgpt add -i 2 -t data -b $start -s $size -l Root $1
+	# $size is in 512 byte blocks while ext4 uses a block size of 1024 bytes
+	mkfs.ext4 -F -b 1024 -m 0 -O ^has_journal -E offset=$(($start * 512)) $1 $(($size / 2))
+}
+
+create_image ${SD_IMG_BASE} 50M 40
+create_image ${SSD_IMG_BASE} 512 30785536
+
+dd if=build/vmlinux.kpart of=${SSD_IMG_BASE} conv=notrunc seek=${P1BYTES}
+dd if=build/vmlinux.kpart of=${SD_IMG_BASE} conv=notrunc seek=${P1BYTES}
+
+mount-FULL -o loop,noatime,offset=${P2STARTBYTES} ${SSD_IMG_BASE} /mnt/ssdimagep2
+mount-FULL -o loop,noatime,offset=${P2STARTBYTES} ${SD_IMG_BASE} /mnt/sdimagep2
+
+mkdir -p /mnt/ssdimagep2/${DISTRO_FILE_PREFIX}-${DISTRO_VERSION}
+for SFS in build/*.sfs; do
+	cp -f ${SFS} /mnt/ssdimagep2/${DISTRO_FILE_PREFIX}-${DISTRO_VERSION}/
+	BASE=${SFS##*/}
+	ln -s ${DISTRO_FILE_PREFIX}-${DISTRO_VERSION}/${BASE} /mnt/ssdimagep2/${BASE}
+done
+mkdir -p ../../local-repositories/frugalify
+case $WOOF_TARGETARCH in
+x86*) FRUGALIFY=frugalify-overlayfs-i386 ;;
+arm|aarch64) FRUGALIFY=frugalify-overlayfs-arm ;;
+esac
+[ ! -f ../../local-repositories/frugalify/${FRUGALIFY} ] && wget --tries=1 --timeout=10 -O ../../local-repositories/frugalify/${FRUGALIFY} https://github.com/dimkr/frugalify/releases/latest/download/${FRUGALIFY}
+install -m 755 ../../local-repositories/frugalify/${FRUGALIFY} /mnt/ssdimagep2/init
+cp -a /mnt/ssdimagep2/${DISTRO_FILE_PREFIX}-${DISTRO_VERSION} /mnt/ssdimagep2/*.sfs /mnt/ssdimagep2/init /mnt/sdimagep2/
+
+case $WOOF_TARGETARCH in
+x86*)
+	dd if=/dev/zero of=${LEGACY_IMG_BASE} bs=50M count=40 conv=sparse
+	parted --script ${LEGACY_IMG_BASE} mklabel msdos
+	parted --script ${LEGACY_IMG_BASE} mkpart primary "" ext4 2048s 100%
+	parted --script ${LEGACY_IMG_BASE} set 1 boot on
+	LOOP=`losetup -Pf --show ${LEGACY_IMG_BASE}`
+	PARTUUID=`blkid -s PARTUUID -o value ${LOOP}p1`
+	mkfs.ext4 -F -b 1024 -m 0 -O ^has_journal ${LOOP}p1
+
+	mkdir -p /mnt/legacyimagep1
+	mount-FULL -o noatime ${LOOP}p1 /mnt/legacyimagep1
+
+	cp -f build/vmlinuz /mnt/legacyimagep1
+	extlinux -i /mnt/legacyimagep1
+	dd if=/usr/lib/EXTLINUX/mbr.bin of=${LOOP}
+	cat << EOF > /mnt/legacyimagep1/extlinux.conf
+DEFAULT puppy
+
+LABEL puppy
+	LINUX vmlinuz
+	APPEND root=PARTUUID=$PARTUUID init=/init rootfstype=ext4 rootwait rw
```

Comment:
@01micko Took some time to figure out how to achieve this, but now I have working initrd-less boot!

The build process for dpup Bullseye produces 2 GB images with the Chrome OS partition layout (ARM + x86_64) and the x86_64 version also produces a legacy BIOS image for Chromebooks that have SeaBIOS.

It boots just fine on "normal" PCs too, puts an overlayfs at / and it's very similar to a frugal installation, but without an initrd.